src/common/components/home/MarketplacePreview.tsx

```diff
@@ -15,45 +17,66 @@ const LoadingPreview = () => {
 };
 
 interface MarketplacePreviewProps {
-  loading: boolean;
-  data?: MarketplacePreviewData;
+  subdomain: string;
 }
 
-const MarketplacePreview: FC<MarketplacePreviewProps> = ({ loading, data }) => {
-  const [showDetails, setShowDetails] = useState(false);
+const MarketplacePreview: FC<MarketplacePreviewProps> = ({ subdomain }) => {
   const { track } = useAnalytics();
 
+  const [showDetails, setShowDetails] = useState(false);
   useEffect(() => setShowDetails(isTouchScreenOnly()), []);
-
   const onMouseEnter = useCallback(() => setShowDetails(true), []);
   const onMouseLeave = useCallback(() => setShowDetails(isTouchScreenOnly()), []);
+
+  const marketplaceQuery = useMarketplacePreviewQuery({
+    variables: {
+      subdomain: subdomain,
+    },
+  });
+  let data: MarketplacePreviewData | undefined = marketplaceQuery?.data?.marketplace
+    ? (marketplaceQuery.data.marketplace as MarketplacePreviewData)
+    : undefined;
+  const loading: boolean = marketplaceQuery?.loading;
+
   const onClickMarketplaceLink = useCallback(() => {
     track('Marketplace Selected', {
       event_category: 'Discovery',
-      event_label: data ? data.subdomain : "unknown-subdomain",
+      event_label: data ? data.subdomain : 'unknown-subdomain',
     });
   }, [data, track]);
 
-  if (loading || !dataAreComplete(data)) {
+  if (loading || !dataAreSufficient(data)) {
     return <LoadingPreview />;
   }
 
-  // all required data are available after checking dataAreComplete()
+  // sufficient data are available after checking dataAreSufficient()
   data = data!;
 
   const marketplaceUrl: string = `https://${data.subdomain}.holaplex.market`;
   const nftVolumeStr: string = Number.parseInt(data.stats.nfts).toLocaleString();
-  const priceSol: number = Number.parseFloat(data.auctionHouse.stats.floor) / LAMPORTS_PER_SOL;
+  const floorPriceSol: number =
+    Number.parseFloat(data.auctionHouse.stats?.floor || '0') / LAMPORTS_PER_SOL;
 
   return (
     <Container onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
       {/* preview image */}
-      <div className="relative h-full w-full">
-        <a href={marketplaceUrl} target="_blank" rel="noreferrer" onClick={onClickMarketplaceLink}>
+      <div className="relative flex">
+        <a
+          href={marketplaceUrl}
+          target="_blank"
+          rel="noreferrer"
+          onClick={onClickMarketplaceLink}
+        >
           <img
-            src={imgOpt(data.bannerUrl, 600)}
+            src={imgOpt(data.bannerUrl, 800)}
             alt={`${data.name}`}
-            className="absolute w-full object-cover"
+            className="min-h-full min-w-full object-cover"
```

Comment:
should this be h-full and w-full?

Reply:
No. The banner aspect ratio doesnt match that of the marketplace preview container. We want to force it to cover the whole thing (with `object-cover`) but to not deform the banner image, thus using (`min-h-full min-w-full`)